staging/src/k8s.io/kubectl/pkg/explain/v2/templates/plaintext.tmpl

```diff
@@ -118,7 +118,7 @@ Takes dictionary as argument with keys:
     {{- else if $resolved.items -}}
         {{- /* If the schema is an array then traverse the array item fields */ -}}
         {{- template "output" (set $nextContext "schema" $resolved.items) -}}
-    {{- else if $resolved.additionalProperties -}}
+    {{- else if and $resolved.additionalProperties (not (eq "bool" (printf "%T" $resolved.additionalProperties))) -}}
```

Comment:
I'd like to merge this PR as it looks good to me. However, we should first verify that it fixes the issue. Could you please add a test case in https://github.com/kubernetes/kubernetes/blob/master/test/cmd/discovery.sh (simple CRD including additionalProperties field) that fails when it runs against the master branch (without your patch) and passes when it runs against your patch.

You can do this;
* Prepare test and temporarily revert your changes (by commenting out them?) and run the CI and see it fails
* Enable your changes and run CI and see it passes.

Reply:
Good idea, i'll add it. Let me figure how testing works there.

Reply:
I think, you can disable your code (permanently remove them by backing them up on your local) and run the new tests (by pushing your commit to this PR) and see that this https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124506/pull-kubernetes-integration/1784840922232524800 fails. After that, you can add back your changes in another commit and after pushing them, pull-kubernetes-integration should pass?

Reply:
Oh i see ^, thanks.  I'll test it out.

Reply:
This is the failed run for reference https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124506/pull-kubernetes-integration/1792623815205851136

Reply:
Thanks. This is exactly like I suggested.